nyx_client/data.py

```diff
@@ -111,29 +112,53 @@ def __init__(self, **kwargs):
     def __str__(self):
         return f"Data({self._title}, {self.url}, {self._content_type})"
 
-    def download(self) -> str | None:
-        """Download the content of the data and populate the class content field.
+    def as_string(self) -> str | None:
+        """Download the content of the data as as string.
 
-        This method attempts to download the content from the data's URL
-        and stores it in the `content` attribute.
+        This method attempts to download the content from the data's URL.
 
         Returns:
             The downloaded content as decoded text or None, if the download fails.
```

Comment:
The user has programmatically no way of knowing why the download failed. I'm not sure that's desired. Since `as_string` and `as_bytes` are new - maybe we should consider this now. (To discuss)

Reply:
(one for later)